src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -1321,21 +1321,86 @@ Assign2PCIdentifier(MultiConnection *connection)
  * format ParsePreparedTransactionName returns false, and true otherwise.
  */
 bool
-ParsePreparedTransactionName(char *preparedTransactionName, int *groupId, int *procId,
-							 uint64 *transactionNumber, uint32 *connectionNumber)
+ParsePreparedTransactionName(char *preparedTransactionName,
+							 int *groupId, int *procId,
+							 uint64 *transactionNumber,
+							 uint32 *connectionNumber)
 {
-	const int expectedFieldCount = 4;
-	int parsedFieldCount = 0;
-	bool nameValid = false;
+	char *ptr = preparedTransactionName;
 
-	parsedFieldCount = sscanf(preparedTransactionName, PREPARED_TRANSACTION_NAME_FORMAT,
-							  groupId, procId, transactionNumber, connectionNumber);
-	if (parsedFieldCount == expectedFieldCount)
+	for (int i=0; i<4; i++)
 	{
-		nameValid = true;
+		ptr = strchr(ptr, '_');
+
+		if (ptr == NULL)
+		{
+			return false;
+		}
+
+		/* step ahead of the current '_' character */
+		++ptr;
+
+		switch (i)
+		{
+			case 0:
+			{
+				*groupId = strtol(ptr, NULL, 10);
```

Comment:
pg_atoi ?

Reply:
Nope, we want to avoid `ereport(ERROR ...` on bad input. If we fail to parse the prepared transaction name it just means it's not a Citus _internal_ one, and we have another processing ahead.